### PR TITLE
Dynamic Radial Menus

### PR DESCRIPTION
**Pull request by Rinary1**
## About the PR

fix icons at radial menu, dynamic changes radius

## Why / Balance

If in radial menu has a lot of icons, it becomes impossible to use it

## Technical details

Edit Draw metod of Radial Container, when player opens emote menu, draw method takes child amount, base radius value, radius increment value and Multiplies the number of icons by the radius increase value and add it to base radius value
## Media

image
image
image
image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: Rinary
 - fix: Fixed radial menus overlapping where there's many icons.

### CODE REVIEW

File: Content.Client/UserInterface/Controls/RadialContainer.cs [hunk 1]
```diff
@@ -68,8 +68,22 @@ public RadialContainer()
...
+
+        Radius = baseRadius + (childCount * radiusIncrement);
+    }
+
     protected override void Draw(DrawingHandleScreen handle)
     {
+		UpdateRadius();
+		
         var children = ReserveSpaceForHiddenChildren ? Children : Children.Where(x => x.Visible);
         var childCount = children.Count();
```
Comment (metalgearsloth) [inline Content.Client/UserInterface/Controls/RadialContainer.cs:89]: You're getting the child count twice and iterating the entire collection, you should be passing this into the method.

You should also document this code.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Client/UserInterface/Controls/RadialContainer.cs [hunk 2]
```diff
@@ -67,11 +67,19 @@ public RadialContainer()
...
-
+	
     protected override void Draw(DrawingHandleScreen handle)
     {
+		
+        const float baseRadius = 100f;
+        const float radiusIncrement = 5f;
+		
+		//takes ammount of icons
         var children = ReserveSpaceForHiddenChildren ? Children : Children.Where(x => x.Visible);
```
Comment (metalgearsloth) [inline Content.Client/UserInterface/Controls/RadialContainer.cs:78]:
```suggestion
		
        var children = ReserveSpaceForHiddenChildren ? Children : Children.Where(x => x.Visible);
```

File: Content.Client/UserInterface/Controls/RadialContainer.cs [hunk 3]
```diff
@@ -67,11 +67,19 @@ public RadialContainer()
...
         var childCount = children.Count();
+		
+		//Multiplies the number of icons by the radius increase value and add it to base radius value
+		Radius = baseRadius + (childCount * radiusIncrement);
```
Comment (metalgearsloth) [inline Content.Client/UserInterface/Controls/RadialContainer.cs:82]:
```suggestion
		
		// Add padding from the center at higher child counts so they don't overlap.
		Radius = baseRadius + (childCount * radiusIncrement);
```

**Review by metalgearsloth — APPROVED** [review]

**Comment by SabatonShip** [comment]
Woah! Thanx alot!

**Pull request merged**